hazelcast/src/main/java/com/hazelcast/util/ByteArrayProcessor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.util;
+
+/**
+ * A byte array processor used to produce a byte[] out of the given one.
+ * Example use case would be when you have an encrypted byte array and want to decrypt it and vice-versa.
+ */
+public interface ByteArrayProcessor {
+
+    byte[] process(final byte[] src);
```

Comment:
Do we really need to have finals in interface definitions? Having finals on local variables I already find questionable due to horizontal noise.. but on interfaces...